packages/core/src/internal/decorators/animate.ts

```diff
@@ -12,25 +12,39 @@ import {
 } from '../motion/interfaces.js';
 import { runPropertyAnimations } from '../motion/utils.js';
 
-// https://www.typescriptlang.org/docs/handbook/decorators.html#class-decorators
+// decorator factory that extends the component constructor to inject animations code into it
 export function animate(config: PropertyDrivenAnimation) {
   // eslint-disable-next-line @typescript-eslint/ban-types
-  return function _DecoratorName<T extends { new (...args: any[]): {} }>(constr: T) {
-    const _constr = constr as any;
-    return (class extends _constr {
-      _animations: any;
+  return function _animationDecorator<T extends { new (...args: any[]): AnimatableElement }>(constructor: T) {
+    return (class extends constructor {
+      private _animationReady = false;
+
+      private _animationDemoMode = false;
+
+      _animations = config;
 
       updated(props: Map<string, any>) {
         super.updated(props);
 
-        const self = (this as unknown) as AnimatableElement & { _animations: PropertyDrivenAnimation };
-        self._animations = config || void 0;
+        // const self = (this as unknown) as AnimatableElement;
+        // self._animations = config || void 0;
 
-        if (!self.hasAttribute(PRIVATE_ANIMATION_STATUS_ATTR_NAME)) {
-          self.setAttribute(PRIVATE_ANIMATION_STATUS_ATTR_NAME, AnimationStatus.ready);
-        } else if (!self.hasAttribute('_demo-mode')) {
-          // ignore if element has the _demo-mode attribute set; _demo-mode is used for docs, static examples, and stories
-          runPropertyAnimations(props, self);
+        // init-style booleans here to skip need for expensive attr lookup
+        if (!this._animationReady && !this.hasAttribute(PRIVATE_ANIMATION_STATUS_ATTR_NAME)) {
+          // this prevents animations from loading onpageload
+          this.setAttribute(PRIVATE_ANIMATION_STATUS_ATTR_NAME, AnimationStatus.ready);
+          this._animationReady = true;
+          return;
+        } else if (this._animationDemoMode) {
```

Comment:
I have a question more or less ?

Why we don't combine the two ifs into one the same way as we do for the above case.

```ts
if (this._animationDemoMode || this.hasAttribute('_demo-mode') { 
  this._animationDemoMode = true;
  return;
}
```
Here the v8 will not do the assignment twice because the internal state is not changed so it won't have a performance penalty, instead, gonna save `if-else-if` code branching?

We could even drop the last `else` block in that case

```ts
if (animation) { 
  return
}

if (preventAnimations) {
  return 
}

runPropertyAnimations(props, this);
```
Maybe I'm missing something it's only reading-and-compiling in my head. 

I think it's more clear what we do with the `else` block and the lot of `if-else` sections, and will require less unit-test for branched sections.

And something else, I like the idea of `_demo_mode` attribute to prevent animations, this will be super helpful for visual testing in the case where this was not an option (dropdown/select/loading and more) - why we don't rename it to something that could be part of the API like `disabledMotionAnimation` something pretty obvious and hard to miss-type ;)  

I think this could be a really good addition to the API, to be honest, I think it could have some added factor to unit-testing as well.

Reply:
I can clean up the conditional a bit. I thought it was easy to follow when I wrote it but I was down in it so... 🤷‍♂️

Thanks for the suggestions.

`_demo-mode` already exists. I didn't add anything new there. I just saved us lookups on the storybook site. The only caveat is that folks (meaning _us_) shouldn't be adding/removing demo-mode willy nilly.

Reply:
`_demo-mode` is new for me ... but I plan to abuse it :) 

I hoped that this will solve one of my issues with `accordion` for example I needed to wait additional `200ms` before snapshotting it and wait for the content inside to be rendered. Now when I think more about it - I see that I was totally wrong about it - and this won't do any difference for me. 

But will help with testing the `loading` animations - I hope so! 

Thanks. I don't have any more questions

![end](https://media.giphy.com/media/RlAwj8AGmvdqThKttZ/giphy.gif)

Reply:
🤣

I'm here for the Perd Hapley memes...